crates/ruff_db/src/vfs/path.rs

```diff
@@ -0,0 +1,97 @@
+use camino::Utf8PathBuf;
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum VfsPath {
```

Comment:
@AlexWaygood I think this could be interesting for you. Ideally, `Vendored` would wrap a `VendoredPathBuf` but I decided not to do this as part of this PR.